pkg/dataobj/internal/dataset/page_reader.go

```diff
@@ -166,6 +171,11 @@ func reuseValuesBuffer(dst []Value, src []Value) []Value {
 }
 
 func (pr *pageReader) init(ctx context.Context) error {
+	// Close any existing reader from a previous pageReader init.
+	if err := pr.Close(); err != nil {
```

Comment:
would it make sense to call Close in reset instead of waiting for the next read?

Reply:
some of the columns might never be read if there are not matching predicates 🤔

Reply:
Good idea 👍 We actually still want to do it here regardless in case we did a seek backwards, but also preemptively closing on Reset makes sense.